Add Flask-Session version identification to Redis backends

vchomakov opened the pull request:
This PR adds client identification to Redis connections created by Flask-Session, allowing Redis operators to see which library is connecting to their servers.

Redis servers support the `lib_name` and `lib_version` parameters (introduced in redis-py) to help operators identify which upstream libraries are connecting. This improves observability and debugging for Redis operators.

Following redis-py best practices, `lib_name` and `lib_version` identify the upstream library (Flask-Session) rather than redis-py itself.

## Implementation

- Added `get_flask_session_version()` utility function to retrieve the Flask-Session version using `importlib.metadata` with fallback to `__version__`
- Added `get_redis_py_version()` utility function to retrieve the redis-py version
- Added `add_redis_version_info()` utility function to set `lib_name="redis-py(flask-session_v<version>)"` and `lib_version=<redis-py-version>`
- Updated `RedisSessionInterface.__init__()` to call `add_redis_version_info()`
- Only sets these parameters if not already provided by the user, ensuring user-provided values are never overridden

## ReadTheDocs Configuration Fix

This PR also fixes the missing Sphinx configuration in `.readthedocs.yaml` that was causing all documentation builds to fail:

- Added `configuration: docs/conf.py` to the `sphinx` section
- This key is now required by ReadTheDocs (see [their blog post](https://blog.readthedocs.com/migrate-configuration-v2/))
- This fix unblocks documentation builds for this PR and all future contributions

## Testing

- Added unit tests in `tests/test_redis.py` to verify version identification behavior
- Tests verify that custom `lib_name` and `lib_version` values are respected and not overridden
- All existing tests pass

## Checklist:

- [x] Add tests that demonstrate the correct behavior of the change. Tests should fail without the change.
- [x] Add an entry in CHANGES.rst summarizing the change and linking to the issue.
- [x] Run pytest and tox, no tests failed.

vchomakov committed:
Add Flask-Session version identification to Redis backends

vchomakov committed:
Fix ReadTheDocs configuration

hasansezertasan commented on src/flask_session/_utils.py:
```diff
@@ -65,3 +65,72 @@ def wrapper(*args: Any, **kwargs: Any) -> Any:
+            return "unknown"
+
+
+def get_redis_py_version() -> str:
```
Do we need this? See:

https://github.com/redis/redis-py/blob/807da5e2b7e650f7711516acd4e535cdf212d0d5/redis/__init__.py#L67

vchomakov replied:
Good catch — we don't need to set `lib_version` at all: redis-py already defaults it to its own installed version, which is exactly what we were setting. flask-session's version is already carried inside `lib_name`. I'll drop the helper and pass only `lib_name`.

hasansezertasan reviewed (changes requested):
It's like user-agent for the redis connection, am I right? Library users can provide their own initialized Redis instance with `lib_name` and `lib_version` provided.

Apart from that, I think it would be better if flask-session would initialize the Redis instance with defaults (if Redis itself doesn't have any default).

hasansezertasan commented on src/flask_session/_utils.py:
```diff
@@ -65,3 +65,72 @@ def wrapper(*args: Any, **kwargs: Any) -> Any:
     return decorator
+
+
+def get_flask_session_version() -> str:
```
Do we need this, see:

https://github.com/pallets-eco/flask-session/blob/bc2fe67958bff5e46023c4807b5e75ca350554eb/src/flask_session/__init__.py#L3

vchomakov replied:
You're right — `__version__` is already the source of truth, so the lookup was redundant. Dropped it, `lib_name` now uses `__version__` directly.

vchomakov committed:
Remove redundant lib_version handling for Redis client identification

redis-py already defaults lib_version to its own installed version,
which is exactly what get_redis_py_version() duplicated. Set only
lib_name; the flask-session version is carried inside it.

vchomakov committed:
Use flask_session.__version__ directly for Redis client identification

The package version is defined in __init__.py and read by flit at
build time (dynamic = ["version"]), so it is the source of truth.
The importlib.metadata lookup was redundant indirection.

hasansezertasan commented on src/flask_session/redis/redis.py:
```diff
@@ -53,7 +53,10 @@ def __init__(
+            # Add version identification for redis-py client
+            kwargs = {}
+            add_redis_version_info(kwargs)
+            client = Redis(**kwargs)
```
What about this and removing other machinery?

```suggestion
            from flask_session import __version__

            client = Redis(lib_name=f"redis-py(flask-session_v{__version__})")
```

vchomakov replied:
Done. It's a single line in `redis.py` now (with the import at the top of the module rather than inline) and `_utils.py` is back to its original state.

vchomakov committed:
Inline Redis client identification and drop the helper

vchomakov committed:
Use DriverInfo on redis-py >= 7.1, fall back to lib_name on older versions
